solvers/snapml.py

```diff
@@ -13,12 +16,14 @@ class Solver(BaseSolver):
 
     def set_objective(self, X, y, lmbd):
         self.X, self.y, self.lmbd = X, y, lmbd
+        self.use_gpu = cuda_version is not None
 
         self.clf = LogisticRegression(
             fit_intercept=False,
             regularizer=self.lmbd,
             penalty="l2",
             tol=1e-12,
+            use_gpu=self.use_gpu,
```

Comment:
The recommendation in the [LogReg doc](https://snapml.readthedocs.io/en/latest/logistic_regression.html) is also to add:
```suggestion
            use_gpu=self.use_gpu,
            dual=self.X.shape[0] >= self.X.shape[1]
```